### PR TITLE
backport in 74X : fix SingleTrackVertexConstraint class for the B=0T case 

### PR DESCRIPTION
**Pull request by HuguesBrun**
backport of the PR #9971 in 74X

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoVertex/KalmanVertexFit/src/SingleTrackVertexConstraint.cc [hunk 1]
```diff
@@ -37,7 +37,8 @@ SingleTrackVertexConstraint::BTFtuple SingleTrackVertexConstraint::constrain(
   typedef VertexTrack<5>::RefCountedLinearizedTrackState RefCountedLinearizedTrackState;
 
   double field  = track.field()->inInverseGeV(track.impactPointState().globalPosition()).z();
-  if (fabs(field) < 1e-4) {
+  double field0  = track.field()->inTesla(GlobalPoint(0.,0.,0.)).z();
```
Comment (VinInn) [inline RecoVertex/KalmanVertexFit/src/SingleTrackVertexConstraint.cc]: here
field().nominalValue() should be used
(mind is integer and in KG)
http://cmslxr.fnal.gov/lxr/source/MagneticField/Engine/interface/MagneticField.h?v=CMSSW_7_5_0_pre5#0056

**Comment by VinInn** [comment]
-1
to test BOff use nominalValue.
for instance
http://cmslxr.fnal.gov/lxr/source/RecoTracker/TkSeedGenerator/plugins/SeedFromConsecutiveHitsCreator.cc?v=CMSSW_7_5_0_pre5#0040

**Comment by HuguesBrun** [comment]
updated as the 76X PR

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test
(trying to get past the infrequent DAS issues)

**Comment by slava77** [comment]
@cmsbuild please test
(DAS error resolution)

**Comment by cvuosalo** [comment]
+1

For #9975 f8c32ff46d8d74a8b316e1d36b84ad3333fb9905

Bug fix to correctly handle SingleTrackVertexConstraint when magnet is off. The 75X and 76X versions of this PR, #9972 and #9971, have already been approved by Reco.

The code change is satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9975/2630/summary.html) against baseline CMSSW_7_4_X_2015-07-01-2300 show no significant differences.

**Comment by slava77** [comment]
To check the impact on pp data with 0T I checked (still run 1) 100 events from run 200781 (PU~30) and processed it with the prompt reco workflow (matrix 1000.0 with the "run1" customizations) in 7_4_4_patch2 (just because a baseline reference file was available there easily).

The result are as expected:
- the main difference is that the STA muons updated at vertex are now there
wf1000_nstaupd
wf1000_staupd_pt

Most of the global muons are unchanged, but there are small variations due to now combinations with "updated" tracks. (the weight of the STA in the track parameters of the global for these events is small, so, the impact on the global track distributions is small).

**Comment by davidlange6** [comment]
+1

**Pull request merged**